google/resumable_media/requests/download.py

```diff
@@ -82,44 +82,67 @@ def _write_to_stream(self, response):
+            # emit the info log `_MISSING_CHECKSUM`.
+            expected_checksum = _get_expected_checksum(
+                response, self._get_headers, self.media_url, checksum_type=checksum
+            )
+        else:
+            expected_checksum = None
 
-        if expected_md5_hash is None:
-            md5_hash = _DoNothingHash()
+        if expected_checksum is not None and checksum == "md5":
+            checksum_object = hashlib.md5()
+        elif expected_checksum is not None and checksum == "crc32c":
+            checksum_object = _helpers._get_crc32c_object()
         else:
-            md5_hash = hashlib.md5()
+            checksum_object = _DoNothingHash()
+
         with response:
-            # NOTE: This might "donate" ``md5_hash`` to the decoder and replace
-            #       it with a ``_DoNothingHash``.
-            local_hash = _add_decoder(response.raw, md5_hash)
+            # NOTE: In order to handle compressed streams gracefully, we try
+            # to insert our checksum object into the decompression stream. If
+            # the stream is indeed compressed, this will delegate the checksum
+            # object to the decoder and return a _DoNothingHash here.
+            local_checksum_object = _add_decoder(response.raw, checksum_object)
             body_iter = response.iter_content(
                 chunk_size=_helpers._SINGLE_GET_CHUNK_SIZE, decode_unicode=False
             )
             for chunk in body_iter:
                 self._stream.write(chunk)
-                local_hash.update(chunk)
+                local_checksum_object.update(chunk)
 
-        if expected_md5_hash is None:
+        if expected_checksum is None:
             return
-
-        actual_md5_hash = base64.b64encode(md5_hash.digest())
-        # NOTE: ``b64encode`` returns ``bytes``, but ``expected_md5_hash``
-        #       came from a header, so it will be ``str``.
-        actual_md5_hash = actual_md5_hash.decode(u"utf-8")
-        if actual_md5_hash != expected_md5_hash:
-            msg = _CHECKSUM_MISMATCH.format(
-                self.media_url, expected_md5_hash, actual_md5_hash
-            )
-            raise common.DataCorruption(response, msg)
-
-    def consume(self, transport):
+        else:
+            actual_checksum = base64.b64encode(checksum_object.digest())
+            # NOTE: ``b64encode`` returns ``bytes``, but ``expected_checksum``
+            #       came from a header, so it will be ``str``.
+            actual_checksum = actual_checksum.decode(u"utf-8")
+            if actual_checksum != expected_checksum:
+                msg = _CHECKSUM_MISMATCH.format(
+                    self.media_url,
+                    expected_checksum,
+                    actual_checksum,
+                    checksum_type=checksum.upper(),
+                )
+                raise common.DataCorruption(response, msg)
+
+    def consume(self, transport, checksum="md5"):
```

Comment:
I think you need to still update this bit to be populated at Init adding a comment to make sure I don't miss in subsequent reviews.